tests/testthat/test-3agglomerate.R

```diff
@@ -120,6 +120,40 @@ test_that("agglomerate", {
     not_uniq <- mergeFeaturesByRank(se, rank = "Species", make_unique = FALSE)
     expect_true( !any( duplicated(rownames(uniq)) ) )
     expect_true( any( duplicated(rownames(not_uniq)) ) )
+    
+    
+    data(GlobalPatterns, package="mia")
+    tse <- GlobalPatterns
+    tse <- transformAssay(tse, assay.type="counts", method="relabundance")
+    altExp(tse, "Family") <- agglomerateByRank(tse, rank="Family")
+    altExp(tse, "Family1") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE)
+    altExp(tse, "Family2") <- agglomerateByRank(tse, rank="Family", onRankOnly = FALSE)
+    altExp(tse, "Family3") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE, na.rm = TRUE)
+    altExp(tse, "Family4") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE, na.rm = FALSE)
+    altExp(tse, "Family5") <- agglomerateByVariable(tse, f="Family", MARGIN = 'row')
+    
+    # Other group is added by agglomerateByPrevalence function to collect features under threshold
+    actual <- agglomerateByPrevalence(tse, rank="Family", assay.type="relabundance", 
+                                         detection  = 0.5/100, prevalence  = 20/100)
+    actual0 <- agglomerateByPrevalence(altExp(tse, "Family"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    actual1 <- agglomerateByPrevalence(altExp(tse, "Family1"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    actual2 <- agglomerateByPrevalence(altExp(tse, "Family2"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    actual3 <- agglomerateByPrevalence(altExp(tse, "Family3"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    actual4 <- agglomerateByPrevalence(altExp(tse, "Family4"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    actual5 <- agglomerateByPrevalence(altExp(tse, "Family5"), assay.type="relabundance", 
+                                       detection  = 0.5/100, prevalence  = 20/100)
+    
+    expect_equal(nrow(actual), nrow(actual0))
+    expect_equal(nrow(actual), nrow(actual1))
+    expect_equal(nrow(actual2), 27)
+    expect_equal(nrow(actual3), 20)
+    expect_equal(nrow(actual), nrow(actual4))
+    expect_equal(nrow(actual), nrow(actual5))
```

Comment:
Add comments: why they are same, and why there are 20 and 27 featues

Reply:
ACK